elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchJson.java

```diff
@@ -123,10 +123,12 @@ private static void visitMappingProperties(Deque<String> path,
     Iterable<Map.Entry<String, JsonNode>> iter = mapping::fields;
     for (Map.Entry<String, JsonNode> entry : iter) {
       final String name = entry.getKey();
-      final ObjectNode node = (ObjectNode) entry.getValue();
-      path.add(name);
-      visitMappingProperties(path, node, consumer);
-      path.removeLast();
+      if (entry.getValue() instanceof ObjectNode) {
+        final ObjectNode node = (ObjectNode) entry.getValue();
+        path.add(name);
+        visitMappingProperties(path, node, consumer);
+        path.removeLast();
+      }
```

Comment:
Do you need `else`?

Reply:
An exception occurs when value node is encountered, but we don't need to handle ValueNode.

Reply:
maybe it's better to restricting visiting nodes to just subtree of `*/properties/` ?

Reply:
thanks for your advice, I have post a new commit.